Sources/StreamChat/StateLayer/ChannelListState+Observer.swift

```diff
@@ -43,5 +95,32 @@ extension ChannelListState {
                 log.error("Failed to start the channel list observer for query: \(query)")
             }
         }
+        
+        // MARK: Linking and Unlinking Channels from the Query
+        
+        private func isBelongingToChannelListQuery(channel: ChatChannel) -> Bool {
+            if let filter = dynamicFilter {
+                return filter(channel)
+            }
+            // When auto-filtering is enabled the channel will appear or not automatically if the
+            // query matches the DB Predicate. So here we default to saying it always belong to the current query.
+            if clientConfig.isChannelAutomaticFilteringEnabled {
+                return true
+            }
+            return false
```

Comment:
Addresses @nuno-vieira [comment](https://github.com/GetStream/stream-chat-swift/pull/3057#discussion_r1514745324) in the previous PR.